File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    '''
    [1,2,3,3] = true
     i
    approach 1 = Counter using a hashmap 
        O(1) lookup and insertion
        O(2n) space for keys and values
    approach 2 = Using a hashset 
        O(1) lookup and insertion
        O(n) space
    '''
    def hasDuplicate(self, nums: List[int]) -> bool:
        hash_set = set()
        for num in nums:
            if num in hash_set:
                return True
            hash_set.add(num)

        return False

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     * 
     * Approach one: pointer counter
     * - pointer, ptr 
     * - maximum value, maxCon - initialized to 0
     * - increment value of ptr, ptr++
     *   - if num[i] === 1
     * - if num[i] === 0 or end of array
     *   - maxCon = max(ptr, maxCon)
     * 
     */
    findMaxConsecutiveOnes(nums: number[]): number {
        let ptr = 0, maxCon = 0;

        for (let i = 0; i < nums.length; i++) {
            if (nums[i] === 1)
                ptr++;
            else {
                maxCon = Math.max(ptr, maxCon);
                ptr = 0;
            }
        }

        maxCon = Math.max(ptr, maxCon);

        return maxCon;
    }
}
